easybuild/easyblocks/o/openblas.py

```diff
@@ -94,3 +168,132 @@ def sanity_check_step(self):
             'dirs': [],
         }
         super(EB_OpenBLAS, self).sanity_check_step(custom_paths=custom_paths)
+
+    def _optarch_for_compiler(self, optarch):
+        """
+        Extracts the appropriate optarch for the compiler currently being used.
+        If it is not compiler-specific it is returned as-is.
+        If no optarch is found, False is returned.
+        :param optarch: A complete optarch statement.
+        https://easybuild.readthedocs.io/en/latest/Controlling_compiler_optimization_flags.html
+        """
+        if optarch is False:
+            return False
+
+        compiler = self.toolchain.comp_family()
+        compiler_specific_optarch_string = ''
+
+        if type(optarch) == str:
+            compiler_specific_optarch_string = optarch
+        elif type(optarch) == dict:
+            if compiler in optarch:
+                compiler_specific_optarch_string = optarch[compiler]
+        else:
+            raise EasyBuildError("optarch in an unexpected format: '%s'" % type(optarch)).__class__.__name__
```

Comment:
Please remove the `.__class__.__name__`. Also maybe include the optarch value and use `,` instead of `%`